x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_base_client.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import assert from 'assert';
+import type { SearchHit, SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import type { Stored } from '../types';
+import type { IndexNameProvider } from './rule_migrations_data_client';
+
+export class RuleMigrationsDataBaseClient {
+  constructor(
+    protected getIndexName: IndexNameProvider,
+    protected username: string,
+    protected esClient: ElasticsearchClient,
+    protected logger: Logger
+  ) {}
+
+  protected processResponseHits<T extends object>(
+    response: SearchResponse<T>,
+    override?: Partial<T>
+  ): Array<Stored<T>> {
+    return this.processHits(response.hits.hits, override);
```

Comment:
please make the final hits value optional `response.hits?.hits` . We had a bug in the past where an async search change broke our api. Our response was `{ hits: {} }` when we expected `{ hits: { hits: [] } }`, leading to our api erroring out before it could get the data. Just some future-proofing 😄

Reply:
`response.hits.hits` should be okay if I make `processHits` to accept _undefined_ for the `hits` param, right?

Reply:
yup!